File: src/main/java/main/commands/slashcommands/commands/dev/ReloadConfigCommand.java
package main.commands.slashcommands.commands.dev;

import main.commands.prefixcommands.CommandContext;
import main.commands.prefixcommands.IDevCommand;
import main.commands.RandomMessageManager;
import main.constants.ENV;
import main.main.Config;
import main.main.Robertify;
import main.utils.component.interactions.AbstractSlashCommand;
import main.utils.locale.LocaleManager;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
import org.jetbrains.annotations.NotNull;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.script.ScriptException;

public class ReloadConfigCommand extends AbstractSlashCommand implements IDevCommand {
    private final static Logger logger = LoggerFactory.getLogger(ReloadConfigCommand.class);

    @Override
    public void handle(CommandContext ctx) throws ScriptException {
        if (!permissionCheck(ctx)) return;

        try {
            Config.reload();
            LocaleManager.reloadLocales();
            Robertify.initVoteSiteAPIs();
            RandomMessageManager.setChance(Double.parseDouble(Config.get(ENV.RANDOM_MESSAGE_CHANCE)));

            ctx.getMessage().addReaction("✅").queue();
        } catch (Exception e) {
            logger.error("There was an unexpected error!", e);
            ctx.getMessage().addReaction("❌").queue();
        }
    }

    @Override
    public String getName() {
        return "reload";
    }

    @Override
    public String getHelp(String prefix) {
        return null;
    }

    @Override
    protected void buildCommand() {
        setCommand(
                getBuilder()
                        .setName("reload")
                        .setDescription("Reload all config files into memory")
                        .setDevCommand()
                        .build()
        );
    }

    @Override
    public String getHelp() {
        return null;
    }

    @Override
    public void onSlashCommand(@NotNull SlashCommandEvent event) {
        if (!devCheck(event))
            return;

        try {
            Config.reload();
            RandomMessageManager.setChance(Double.parseDouble(Config.get(ENV.RANDOM_MESSAGE_CHANCE)));
            event.reply("Successfully reloaded all configs!").setEphemeral(true).queue();
        } catch (Exception e) {
            logger.error("There was an unexpected error!", e);
            event.reply("There was an unexpected error!\n```" + e.getMessage() + "```")
                    .setEphemeral(true)
                    .queue();
        }
    }
}

File: src/main/kotlin/main/commands/slashcommands/dev/ReloadConfigCommand.kt
package main.commands.slashcommands.dev

import dev.minn.jda.ktx.util.SLF4J
import main.main.Config
import main.main.Robertify
import main.utils.RobertifyEmbedUtils.Companion.sendEmbed
import main.utils.component.interactions.slashcommand.AbstractSlashCommand
import main.utils.component.interactions.slashcommand.models.SlashCommand
import main.utils.locale.LocaleManager
import main.utils.managers.RandomMessageManager
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent

class ReloadConfigCommand : AbstractSlashCommand(SlashCommand(
    name = "reload",
    description = "Reload all config files.",
    developerOnly = true
)) {

    companion object {
        private val logger by SLF4J
    }

    override fun handle(event: SlashCommandInteractionEvent) {
        val guild = event.guild!!
        event.deferReply(true).queue()

        try {
            Config.reload()
            LocaleManager.reloadLocales()
            Robertify.initVoteSiteAPIs()
            RandomMessageManager.chance = Config.RANDOM_MESSAGE_CHANCE

            event.hook.sendEmbed(guild, "Successfully reloaded all configs!")
                .queue()
        } catch (e: Exception) {
            logger.error("There was an unexpected error!", e)
            event.hook.sendEmbed(guild, "There was an unexpected error!\n```${e.message ?: "No message provided."}```")
                .queue()
        }

    }
}